### PR TITLE
ddl: add a binlog when dropping column

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
Drainer receives a binlog without this column of data when the job is in the "Delete only" state. In this case, if we don't do special treatment, Drainer will think that this binlog has one less column.

### What is changed and how it works?
We add this binlog to enable Drainer to handle DML correctly in "Delete only" and "Delete reorg".
Related to pingcap/tipb#156

### Check List 

Tests 
 - No code

Related changes

 - Need to cherry-pick to the release branch

Release note

 - We add this binlog to enable Drainer to handle DML correctly in "Delete only" and "Delete reorg".

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @crazycs520 @bb7133 @suzaku

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -191,17 +191,18 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
...
 
 	ddlQuery = AddSpecialComment(ddlQuery)
 	info := &BinlogInfo{
 		Data: &binlog.Binlog{
-			Tp:       binlog.BinlogType_Prewrite,
-			DdlJobId: jobID,
-			DdlQuery: []byte(ddlQuery),
+			Tp:             binlog.BinlogType_Prewrite,
+			DdlJobId:       jobID,
+			DdlSchemaState: ddlSchemaState,
```
Comment (winkyao) [inline sessionctx/binloginfo/binloginfo.go]: What about the compatibility?

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -462,6 +461,16 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 	}
 }
 
+func writeBinlog(binlogCli *pumpcli.PumpsClient, txn kv.Transaction, job *model.Job) {
+	if job.IsDone() || job.IsRollbackDone() ||
+		// Drainer receives a binlog without this column of data when the job is in the "Delete only" state.
```
Comment (bb7133) [inline ddl/ddl_worker.go]:
```suggestion
		// Drainer receives a binlog without this column of data when the job is in the "DeleteOnly" state.
```
How about "DeleteOnly" or "delete only"? BTW the comment sounds awkward, can you refine it?

**Comment by bb7133** [comment]
rest LGTM

File: ddl/ddl_worker.go (hunk 2 above)
Comment (zimulala) [reply ddl/ddl_worker.go]: Any suggestions for this comment?

**Pull request merged**